### ISSUE TITLE
Please export EtaError type

### ISSUE THREAD

**Issue by xuxucode** [eta-dev/eta#243]
**Is your feature request related to a problem? Please describe.**

We are using eta to render message codes (as named templates) to localized messages. Since there many message codes exist, lazy loading templates is decided. For example:

```js
try {
  eta.render('@username_already_exists', { username: 'something' })
} catch (err) {
  if (err instanceof EtaError) { // Cannot find name 'EtaError'
                        ^
    eta.loadTemplate('@username_already_exists', '<%= it.username %> already exists.')
  }
  // render again
}
```

**Describe the solution you'd like**

Please export the `EtaError`, or export a more specific error, `EtaTemplateNotFoundError` for example.

**Comment by nebrelbug** [comment]
@xuxucode this seems like a good idea which I'll try to release soon. Do you think it would be helpful to have separate errors for different kinds of bugs, or is it sufficient to have just one exported error type?

**Comment by xuxucode** [comment]
Thanks. It's helpful to have separate errors, my 2 cents:

- `EtaFileTempalteError`: views not set, file path cannot be resolved, file not found
- `EtaNamedTempalteError`: named template not defined(loaded), 
- `EtaParseError`: parsing template content failed
- `EtaRuntimeError`: runtime errors

**Comment by nebrelbug** [comment]
@xuxucode thanks for the input! I'll wait for a few days before working on this to see if others have feedback.

**Comment by nebrelbug** [comment]
I think this is a good idea. Desired behavior should be that Eta throws 4 different errors, each extended from one base `Eta` class:

- `EtaFileResolutionError`
- `EtaNameResolutionError`
- `EtaParseError`
- `EtaRuntimeError`

I don't have time to implement this right now, but anyone is welcome to submit a PR with this behavior.

**Comment by nebrelbug** [comment]
Closed by #276 :rocket:
